### PR TITLE
Encapsulate TruffleLanguage.Env in EnsoContext

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Let's control the services that are exposed from `TruffleLanguage.Env` by hiding the reference in `private` field and only exposing just the necessary methods. This PR just directly exposes everything that was needed, but further work shall remove some of the exposed methods - the first candidate being `asGuestValue`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
- All code has been tested:
  - [x] Unit tests continue to pass

### CODE REVIEW

File: engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/interop/generic/GetSourceLocationNode.java [hunk 1]
```diff
@@ -24,7 +24,6 @@ public class GetSourceLocationNode extends Node {
   Object execute(Object value) {
     try {
       return EnsoContext.get(this)
-          .getEnvironment()
           .asGuestValue(new EnsoSourceSection(library.getSourceLocation(value)));
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/interop/generic/GetSourceLocationNode.java:27]: Future work: We shouldn't use `asGuestValue` to wrap `EnsoSourceSection`. Rather `EnsoSourceSection` shall implement `EnsoObject` and `@ExportMessage`s directly.

File: engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EnsoProjectNode.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ public Object getOtherProjectDescr(
 
   private static Atom createProjectDescriptionAtom(EnsoContext ctx, Package<TruffleFile> pkg) {
     EnsoFile rootPath = new EnsoFile(pkg.root().normalize());
-    Object cfg = ctx.getEnvironment().asGuestValue(pkg.getConfig());
+    Object cfg = ctx.asGuestValue(pkg.getConfig());
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EnsoProjectNode.java:129]: Future work: Exposing `pkg.getConfig()` is wild idea - we should create an `EnsoObject` subclass and expose the needed members via `@ExportMessage`.

File: engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java [hunk 3]
```diff
@@ -405,6 +413,45 @@ public void addToClassPath(TruffleFile file) {
...
+  public Object asJavaPolyglotObject(Object obj) {
+    return environment.asHostObject(obj);
+  }
+
+  /**
+   * Wraps a Java object into interop object.
+   * @param obj java object
+   * @return wrapper object
+   */
+//  @Deprecated(forRemoval=true)
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java:450]: Future work: remove the need for this method.

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/Builtin.java [hunk 4]
```diff
@@ -240,7 +240,7 @@
    *     try {
    *       return self.create(path)
    *     } catch (java.io.IOException e) {
-   *       throw new PanicException(EnsoContext.get(this).getEnvironment().asGuestValue(e), this);"
+   *       throw new PanicException(EnsoContext.get(this).asGuestValue(e), this);"
```
Comment (JaroslavTulach) [inline lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/Builtin.java:243]: Future work: there is a lot of usages of `asGuestValue` in the generated code. Usually it catches an exception and converts it into a guest value which is then used as a `payload` of `PanicException`. Can we get rid of that and just generate `PanicException` with some meaningful `Atom` as a `payload`?

**Review by hubertp — APPROVED** [review]
Nice!

File: engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsComplexNode.java [hunk 5]
```diff
@@ -417,7 +417,7 @@ boolean equalsInteropObjectWithMembers(
     }
   }
 
-  @Specialization(guards = {"isHostObject(selfHostObject)", "isHostObject(otherHostObject)"})
+  @Specialization(guards = {"isJavaObject(selfHostObject)", "isJavaObject(otherHostObject)"})
```
Comment (Akirathan) [inline engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsComplexNode.java:420]: I find this renaming a bit unintuitive. `JavaObject` can be both *host* and *guest* object (depending on whether the object was created in Espresso). No? Shouldn't we keep the old naming?

**Review by Akirathan — APPROVED** [review]
Good idea for refactoring. I just have problems understanding the new nomenclature with `Java` vs `Host` vs `Guest`. Can you explain that further in my comments, pls?

File: engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsComplexNode.java (hunk 5 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsComplexNode.java:420]: Espresso runs as a guest language, and as such it produces  _"guest objects"_. Depending on whether we use Enso regular Java interop or Espresso interop the definition of `isJavaObject` changes - either it checks for host objects or it checks for objects coming from Espresso. That's why I was searching for some uniform name and came up with `isJavaObject`.

True, we don't want Java object from the interpreter, just Java object from regular or Espresso interop. So I agree the naming isn't perfect.

**Pull request merged**